src/files/BrsFile.ts

```diff
@@ -183,13 +183,15 @@ export class BrsFile {
     /**
      * Walk the AST and find the expression that this token is most specifically contained within
      */
-    public getClosestExpression(position: Position) {
+    public getClosestExpression(position: Position | Range) {
         const handle = new CancellationTokenSource();
         let containingNode: Expression | Statement;
         this.ast.walk((node) => {
             const latestContainer = containingNode;
             //bsc walks depth-first
-            if (util.rangeContains(node.range, position)) {
+            if (isPosition(position) && util.rangeContains(node.range, position)) {
+                containingNode = node;
+            } else if (isRange(position) && util.rangeContainsRange(node.range, position)) {
```

Comment:
This seems a bit wasteful to run the `isPosition` and `isRange` call for every ast walk iteration. Consider caching the values at the top of the function?

```typescript
const paramIsPosition = isPosition(position);
const paramIsRange = isRange(position);
```

Reply:
just removed the whole thing here... not needed